src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -114,22 +117,27 @@ private static Type MaxKind(Type a, Type b)
         /// <param name="guessRows">number of rows used to guess types</param>
         /// <param name="addIndexColumn">add one column with the row index</param>
         /// <returns>DataFrame</returns>
-        public static DataFrame ReadStream(Func<StreamReader> createStream,
+        public static DataFrame LoadCsv(Stream csvStream,
                                 char separator = ',', bool header = true,
                                 string[] columnNames = null, Type[] dataTypes = null,
                                 long numberOfRowsToRead = -1, int guessRows = 10, bool addIndexColumn = false)
         {
+            if (!csvStream.CanSeek)
+                throw new ArgumentException(Strings.NonSeekableStream, nameof(csvStream));
+
             var linesForGuessType = new List<string[]>();
             long rowline = 0;
             int numberOfColumns = 0;
 
             if (header == true && numberOfRowsToRead != -1)
                 numberOfRowsToRead++;
 
+            List<DataFrameColumn> columns;
+            long streamStart = csvStream.Position;
             // First pass: schema and number of rows.
-            using (var st = createStream())
+            using (var streamReader = new StreamReader(csvStream, Encoding.UTF8, detectEncodingFromByteOrderMarks: true, bufferSize: 1024, leaveOpen: true))
```

Comment:
I believe the right defaults for the other parameters should be:

```C#
StreamReader(Stream stream, [NullableAttribute(2)] Encoding? encoding = null, bool detectEncodingFromByteOrderMarks = true, int bufferSize = -1, bool leaveOpen = false);
```

At least that is what `netcoreapp3.0` has for defaults on these parameters.

Reply:
Weird, I don't see the constructor with the default values in VS, but I see it on source.dot.net. I'm on netstandard 2.0

Reply:
The constructor with the default values isn't on `netstandard2.0`. It is only on `netcoreapp3.0`.

The reason I brought it up is because you are forced to put default values here, and you should be using the same defaults as we added for `netcoreapp3.0`.